src/features/user/userSlice.js

```diff
@@ -0,0 +1,54 @@
+import { createSlice } from "@reduxjs/toolkit";
+import {
+  editUser,
+  followUser,
+  getAllUsers,
+  unfollowUser,
+} from "../../helper/user";
+
+const initialState = {
+  users: [],
+};
+
+export const userSlice = createSlice({
+  name: "user",
+  initialState,
+  extraReducers: {
+ 
+    [getAllUsers.fulfilled]: (state, { payload }) => {
+      state.users = payload;
+    },
+
+    [followUser.fulfilled]: (state, { payload }) => {
+      state.users = state.users.map((user) => {
+        if (user.username === payload.followUser.username)
+          return payload.followUser;
+        if (user.username === payload.user.username) return payload.user;
+        return user;
+      });
+    },
+    [unfollowUser.pending]: (state, { payload }) => {
+      console.log("pending",payload)
+      
+    },
+    [unfollowUser.reject]: (state, { payload }) => {
+      console.log("reject",payload)
```

Comment:
You can remove these two states if you are not doing any operation like loader or showing errors. Else, you can put a TODO above it if you are planning to do it later.